SDWebImage/Core/SDWebImageManager.m

```diff
@@ -167,13 +150,7 @@ - (nullable NSString *)cacheKeyForURL:(nullable NSURL *)url context:(nullable SD
     }
     
     // Transformer Key Appending
-    id<SDImageTransformer> transformer = self.transformer;
-    if (context[SDWebImageContextImageTransformer]) {
-        transformer = context[SDWebImageContextImageTransformer];
-        if (![transformer conformsToProtocol:@protocol(SDImageTransformer)]) {
```

Comment:
This is the *one* place that might have a functional change, but anecdotally I know `conformsToProtocol` is *very* expensive. It's best to avoid this.

Reply:
This may be OK. I didn't actually see anyone store wrong data type in this option.

But however, we already has so many of protocol check, like that `SDAnimatedImageProtocol` check